### PR TITLE
Report Section: Test regressions all

### PR DESCRIPTION
**Pull request by Crola1702**
As the title says. This is the implementation of test_regressions_all section discussed in: https://github.com/osrf/buildfarm-tools/issues/57#issuecomment-2176897854

### CODE REVIEW

File: database/scripts/generate_report.rb [hunk 1]
```diff
@@ -25,20 +25,23 @@
 options[:report_name] = "buildfarm-report_#{DateTime.now.strftime("%Y-%m-%d_%H-%M")}.json" if options[:report_name].nil?
 
 def generate_report(report_name, exclude_set)
+    report_regressions_all = BuildfarmToolsLib::test_regressions_all(filter_known: true)
+    report_regressions_consecutive = BuildfarmToolsLib::test_regressions_today(filter_known: true, only_consistent: true, group_issues: true, report_regressions: report_regressions_all)
+    report_regressions_flaky = BuildfarmToolsLib::flaky_test_regressions(filter_known: true, group_issues: true, report_regressions: report_regressions_all)
+
```
Comment (Blast545) [inline database/scripts/generate_report.rb:31]: Why was this moved outside the report creation below?

**Review by Blast545 — APPROVED** [review]
Can you paste a generated report with this format so we can evaluate its form as a whole?

**Comment by Crola1702** [comment]
> Can you paste a generated report with this format so we can evaluate its form as a whole?

Ahhh I forgot to add it. Good catch

* [report.json](https://github.com/user-attachments/files/16871807/report.json)
* :scroll: [report.md](https://github.com/user-attachments/files/16871806/report2.md)

File: database/scripts/generate_report.rb (hunk 1 above)
Comment (Crola1702) [reply database/scripts/generate_report.rb:31]: `test_regressions_all` method returns every test regression that happened in the last build. `test_regressions_today` and `flaky_test_regressions` run that method internally.

This is an optimization to not fetch all test regressions twice, but use only one run for both
